scripts/tests_markdown.py

```diff
@@ -0,0 +1,306 @@
+        __inject_statistics(endpoint_metrics['sql_time'])
+        __inject_statistics(endpoint_metrics['sql_queries'])
+
+    for _, endpoint_metrics in internals.items():
+        __inject_statistics(endpoint_metrics['total_time'])
+
+    environment['raw_data']['api_endpoint_metrics'] = api_endpoints
+    environment['raw_data']['internals_metrics'] = internals
+
+
+def __inject_raw_timings(environment):
+    all_timings = []
+
+    for test in environment['raw_data']['tests']:
+        if 'metadata' not in test or 'local_metrics' not in test['metadata']:
+            continue
+
+        metrics = test['metadata']['local_metrics']
+        timing = metrics['timing']
+
+        for endpoint, timings in timing.items():
+            recording = timings[0].copy()
+            recording['endpoint'] = endpoint
+            all_timings.append(recording)
+
+    environment['raw_data']['all_timings'] = all_timings
+
+
+def _ensure_has_dict_at_key(the_dict, key, **kwd):
+    if key not in the_dict:
+        the_dict[key] = dict(**kwd)
+
+
+def __inject_api_timing_summary_test(test):
+    metrics = test['metadata']['local_metrics']
+    timing = metrics['timing']
+    counter = metrics['counter']
+    api_endpoints = {}
+    internal_timings = {}
+
+    def summarize_times(timings):
+        times = list(map(lambda t: t['time'], timings))
+        return __inject_statistics({
+            'raw': times,
+        })
+
+    def summarize_counter(c):
+        counters = list(map(lambda t: t['n'], c))
+        return __inject_statistics({
+            'raw': counters,
+        })
+
+    for endpoint, timings in timing.items():
+        if not endpoint.startswith("api"):
+            continue
+
+        endpoint_summary = {
+            'total_time': summarize_times(timings),
+            'label': endpoint[len("api."):]
+        }
+        sql_times = "sql.%s" % endpoint
+        if sql_times in timing:
+            endpoint_summary['sql_time'] = summarize_times(timing[sql_times])
+        sql_queries = "sqlqueries.%s" % endpoint
+        if sql_queries in counter:
+            endpoint_summary['sql_queries'] = summarize_counter(counter[sql_queries])
+        api_endpoints[endpoint] = endpoint_summary
+
+    for endpoint, timings in timing.items():
+        if not endpoint.startswith("internals"):
+            continue
+
+        internal_summary = {
+            'total_time': summarize_times(timings),
+            'label': endpoint[len("internals."):]
+        }
+        internal_timings[endpoint] = internal_summary
+
+    test['api_endpoint_metrics'] = api_endpoints
+    test['internals_metrics'] = internal_timings
+
+
+def __inject_statistics(from_dict):
+    raw_values = from_dict['raw']
+    from_dict['sum'] = sum(raw_values)
+    from_dict['median'] = statistics.median(raw_values)
+    if len(raw_values) > 1:
+        from_dict['stdev'] = statistics.stdev(raw_values)
+    else:
+        from_dict['stdev'] = "n/a"
```

Comment:
```suggestion
        from_dict['stdev'] = None
```
To accommodate float formatting of statistics in the templates (see suggesion below). Also, that way we leave the choice of *how* to display stuff to the template, which, I think, is the appropriate place for it anyway.